### DISCUSSION TITLE
Upgrade to Tesla Version 2023.38.1 - Teslamate shows Model 3 no longer available

### DISCUSSION THREAD

**Discussion by Steffi100** [teslamate-org/teslamate#3370] [Q&A]
After i upgraded my Model 3 to Version: 2023.38.1, Teslamate shows the car in Status: unavailabe, I made sure that the new Location Sharing feature was turned on and the icon appears when i query the car via the Tesla Mobile App (Controls -> Safety -> Allow Mobile Access), However Teslamate is no longer finding the car. 

Any ideas what is happening and how to fix this?

**Comment by jsight** [comment]
There must be a new API location for the location data.

**Reply by yao8086** [reply]
I got the same problem,teslamate shows unavailabe.when login again via token,shows token invalid.

**Comment by NirKli** [comment]
@Steffi100 @yao8086 
Very interesting. Could one of you please attach the Teslamate logs here?

**Comment by jsight** [comment]
I see a lot of this:

```
2023-10-19 05:12:27.818 [error] GenStateMachine :"1" terminating
** (MatchError) no match of right hand side value: {:error, #Ecto.Changeset<action: :insert, changes: %{date: ~U[2023-10-19 05:12:26.920000Z], power: 0, battery_heater: false, driver_temp_setting: Decimal.new("22.0"), fan_status: 0, inside_temp: Decimal.new("20.0"), is_climate_on: false, is_front_defroster_on: false, is_rear_defroster_on: false, outside_temp: Decimal.new("18.5"), passenger_temp_setting: Decimal.new("22.0"), odometer: 1, battery_heater_on: false, battery_level: 42, usable_battery_level: 42, tpms_pressure_fl: Decimal.new("2.6"), tpms_pressure_fr: Decimal.new("2.525"), tpms_pressure_rl: Decimal.new("2.625"), tpms_pressure_rr: Decimal.new("2.575"), ideal_battery_range_km: Decimal.new("381.25"), rated_battery_range_km: Decimal.new("381.25"), est_battery_range_km: Decimal.new("344.98")}, errors: [latitude: {"can't be blank", [validation: :required]}, longitude: {"can't be blank", [validation: :required]}], data: #TeslaMate.Log.Position<>, valid?: false>}
    (teslamate 1.27.3) lib/teslamate/vehicles/vehicle.ex:721: anonymous fn/4 in TeslaMate.Vehicles.Vehicle.handle_event/4
    (ecto_sql 3.10.1) lib/ecto/adapters/sql.ex:1203: anonymous fn/3 in Ecto.Adapters.SQL.checkout_or_transaction/4
    (db_connection 2.5.0) lib/db_connection.ex:1630: DBConnection.run_transaction/4
    (teslamate 1.27.3) lib/teslamate/vehicles/vehicle.ex:713: TeslaMate.Vehicles.Vehicle.handle_event/4
    (stdlib 5.0.1) gen_statem.erl:1379: :gen_statem.loop_state_callback/11
    (stdlib 5.0.1) proc_lib.erl:241: :proc_lib.init_p_do_apply/3
```

**Reply by NirKli** [reply]
Yes, it seems that the API doesn't send the location of the car. It tries to create a position for the car, but the values are missing.
Also, the odometer returns a value of 1, which is a little bit weird.

**Reply by jsight** [reply]
I changed the odometer value. :) It is a reasonable number in the actual log.

**Reply by NirKli** [reply]
Are you sure you enabled location sharing on your Tesla?
![image](https://github.com/adriankumpf/teslamate/assets/45952578/8b7811c7-a43d-40e7-a41b-a3399bd6a671)

**Reply by Steffi100** [reply]
I can query the correct location of the car with the Tesla Mobile App and it shows a red Arrow of the exact location which is confirmation that  the setting: (Allow Mobile Access) is enabled, however if i query the car with teslapy the "drive_state" parameters do no longer return any of the latitude or longitude parameters like "active_route_latitude" or "native_latitude" or "gps_as_of" these parameters are no longer provided from Tesla. In the vehicle query it also says the "api_version": 67, .....i know it still worked with "api_version": 63 (I guess that was 2023.32.7)

**Comment by djpalmis** [comment]
Shall I avoid to install that update? Any other report of this issue?

**Reply by jsight** [reply]
If you can avoid it, yes. It seems to be causing a lot of issues.

**Reply by Steffi100** [reply]
Upgraded to 2023.38.4 today, same behaviour:
![M3NA](https://github.com/adriankumpf/teslamate/assets/148391710/6f4364ef-dfee-45c3-adeb-50d5c456d562)

**Reply by NirKli** [reply]
Issues like what? Can you provide more details about the issues encountered after applying the changes in your PR?

**Reply by marcbaier** [reply]
I also have the same issue. No data in the Teslamte Dashbaord after the software update to 2023.38.4. 
<img width="993" alt="Screen Shot 2023-10-30 at 14 52 56 PM" src="https://github.com/adriankumpf/teslamate/assets/97667985/dce4a8e2-577e-4d0a-bf5b-88ecd5b3a395">

**Reply by wooter** [reply]
A workaround is available in https://github.com/adriankumpf/teslamate/issues/3371

**Reply by jsight** [reply]
@NirKli I was getting a lot of disconnects that were causing a weird issue with logging drives and charging sessions. The latest version of that MR should be working now, though. It has been fine for me for the last 12 hours or so.

**Reply by Steffi100** [reply]
same behaviour with Version 2023.38.6, Will there be an official Teslamate upgraded release to incorporating the suggested workaround?

**Reply by ktsenvolt** [reply]
https://github.com/teslamate-org/teslamate/releases/tag/v1.27.4

**Comment by Steffi100** [comment]
Tahnk you @adriankumpf for New Release 1.27.4: works great. You are the best

**Comment by marcbaier** [comment]
Hi all - i just updated my teslamate docker installation via docker-compose up -d and I don't see the new "battery health" dashboard that should be included. Is there a way I can check if am really on the latest release 1.27.4 ? Or do I have to somehow import the new dashboard into grafana ? Thanks so much for any help !

**Reply by ktsenvolt** [reply]
Update to newer version teslamate:

docker-compose pull
docker-compose up -d

**Reply by marcbaier** [reply]
Thanks a lot !
